src/components/layout/organize/SingleViewLayout.tsx

```diff
@@ -1,25 +1,56 @@
 import { Box } from '@material-ui/core';
 import { FunctionComponent } from 'react';
-import { useQuery } from 'react-query';
+import { useIntl } from 'react-intl';
 import { useRouter } from 'next/router';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
 
+import EditTextinPlace from 'components/EditTextInPlace';
 import getView from 'fetching/views/getView';
+import patchView from 'fetching/views/patchView';
 import TabbedLayout from './TabbedLayout';
 import ViewJumpMenu from 'components/views/ViewJumpMenu';
+import ZetkinQuery from 'components/ZetkinQuery';
+import { ZetkinView } from 'types/views';
 
 
 const SingleViewLayout: FunctionComponent = ({ children }) => {
+    const intl = useIntl();
     const router = useRouter();
+    const queryClient = useQueryClient();
     const { orgId, viewId } = router.query;
     const viewQuery = useQuery(['view', viewId ], getView(orgId as string, viewId as string));
 
-    const view = viewQuery.data;
+    const patchViewMutation = useMutation(patchView(orgId as string, viewId as string));
+
+    const updateTitle = async (view: ZetkinView, newTitle: string) => {
+        try {
```

Comment:
Here you could use NProgress to show the loading state of the update, and you could use an MUI Snackbar to show an error if it fails.

Reply:
- The update is IMO not slow enough to warrant a progress bar, and there are enough visual cues already if you try the component out
- There's a snackbar is in the EditTextInPlace component - we could move it to the parent component, but I don't see a good reason. Think this kind of thing should probably be handled centrally by some notification provider, and some consistent way of display notifications, but this is a quick short term solution.